plugins/filter/from_xml.py

```diff
@@ -0,0 +1,145 @@
+#
+# -*- coding: utf-8 -*-
+# Copyright 2021 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+#
+
+"""
+The from_xml filter plugin
+"""
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+    name: from_xml
+    author: Ashwini Mhatre (@amhatre)
+    version_added: "2.0.2"
+    short_description: convert given xml string to native python dictionary.
+    description:
+        - This plugin converts the xml string to native python dictionary.
+        - Using the parameters below- C(data|ansible.utils.from_xml)
+    options:
+      data:
+        description:
+        - The input xml string .
+        - This option represents the xml value that is passed to the filter plugin in pipe format.
+        - For example C(config_data|ansible.utils.from_xml), in this case C(config_data) represents this option.
+        type: str
+        required: True
+      engine:
+        description:
+        - Conversion library to use within the filter plugin.
+        type: str
+        default: xmltodict
+"""
+
+EXAMPLES = r"""
+
+#### Simple examples with out any engine. plugin will use default value as xmltodict
+
+tasks:
+  - name: convert given xml to native python dictionary
+    ansible.builtin.set_fact:
+      data: "
+        <netconf-state xmlns=\"urn:ietf:params:xml:ns:yang:ietf-netconf-monitoring\"><schemas><schema/></schemas></netconf-state>
+            "
+
+  - debug:
+      msg:  "{{ data|ansible.utils.from_xml }}"
+
+##TASK######
+# TASK [convert given xml to json] *****************************************************************************************************
```

Comment:
Task name doesn't seem to match the task name above?

Reply:
ack